### ISSUE TITLE
Homespan working in homeapp but wont connect to homeKit.

### ISSUE THREAD

**Issue by Gsisafk** [HomeSpan/HomeSpan#940]
I Made a home automation  device using homespan the  device works in the home app but not when i tryto make my Ipad as the 
 homekit device that means i can only  use it when i am connected to the same Wifi as the Esp device for reference i used this https://www.youtube.com/watch?v=K1xrCyN9T98. I want to control the device from  outside the home is it possible for reference my ipadOs is 17.6.1.

**Comment by HomeSpan** [comment]
Apple [requires the use of a Home Hub](https://support.apple.com/en-us/HT207057) for operation outside your WiFi network.

**Comment by Gsisafk** [comment]
I know using the ipad 10th gen as homehub

**Comment by HomeSpan** [comment]
iPad cannot be used as a Home Hub.  Only Apple TV and Home Pod.  Apple discontinued use of iPad as Home Hub about 2-3 years ago.

**Comment by HomeSpan** [comment]
I think I lost the gist of this.  You have successfully connected a HomeSpan device to your Home App on an iPhone by pairing it, correct?  And it works as expected when you operate from the iPhone, correct?  But when you open your iPad to use it, it does not work.  If that is the issue, I have no idea how to address.   HomeKit handles all of this behind the scenes - it's not related to HomeSpan.  Using an iPad as a Home Hub is only possible if you never upgraded to the new HomeKit architecture way back in iOS16.   No idea how the old architecture may or may not operate.

**Comment by Gsisafk** [comment]
No it works on my ipad perfectly as well but only when i am connected to the same wifi as the esp32 when i change the wifi network it stops working. I know that means i need a homehub so i did some research it said my ipad 10th gen with iPadOS 17 can act as one, the site told me to enable home in icloud on my ipad so i went ahead and did that still didnt work so i opened the home app and then the device i added (esp32) then settings it said under homekit certified -NO. all i want to do is control the relays on a different wifi network. 
In short the device works perfectly when used under the same network but does not work under a different network. Thanks for the reply btw

**Comment by HomeSpan** [comment]
Thanks for the clarifications.  The device that is acting as a home hub must be on the same WiFi network as your HomeKit device.  If the iPad is correctly operating as a Home Hub then this means you can use your iPhone to connect back to the HomeKit device via the iPad/Home Hub even if the iPhone is not on the same WiFi network (or you are accessing outside your home via 5G, etc.).  But if you change the network of the iPad/Home Hub, you will lose contact with the device.

**Comment by Gsisafk** [comment]
Yes i know thats the problem the ipad stays connected to the same wifi that further connects to icloud and therefore i can access it via my phone when i am on a different network i know how it works. Thats the functionality thats not working the homekit is not working i did leave the ipad connected to the same wifi. And then only i turned off my wifi on my mobile phone switched to a 5G network and then tried to access the device from my iphone again it said the device was unresponsive on the iphone (conected to mobile data ) and still responsive on the ipad thats  connected to the same wifi still

**Comment by HomeSpan** [comment]
I think that means the iPad is not operating properly as a Home Hub. Do you have any commercial HomeKit devices to compare with?

**Comment by HomeSpan** [comment]
Any update on this, or can I close for now?

**Comment by Gsisafk** [comment]
Thanks it was an apple update they removed the ipad support for homehub nothing to do with the library
